core/src/main/scala/chisel3/internal/Error.scala

```diff
@@ -178,18 +178,26 @@ private[chisel3] object ErrorLog {
 
 private[chisel3] class ErrorLog {
 
+  def getLoc: String = {
+    getUserLineNumber match {
+      case Some(elt: StackTraceElement) => s"${elt.getFileName}:${elt.getLineNumber}"
+      case None => "(unknown)"
+    }
+  }
+
   /** Log an error message */
   def error(m: => String): Unit =
-    errors += new Error(m, getUserLineNumber)
+    errors += (((m, getLoc), new Error(m, getUserLineNumber)))
```

Comment:
It would be better if we didn't call `getUserLineNumber` here twice since it involves getting a stack trace (which is expensive). Can `getLoc` instead accept `Option[StackTraceElement]` and just do the filtering on that and then we can call `getUserLineNumber` just once?